au/code/au/utility/test/probable_primes_test.cc

```diff
@@ -200,6 +200,115 @@ TEST(MillerRabin, SupportsConstexpr) {
     static_assert(result == PrimeResult::PROBABLY_PRIME, "997 is prime");
 }
 
+TEST(IsPerfectSquare, ProducesCorrectAnswers) {
+    auto next_sqrt = 0u;
+    for (auto n = 0u; n < 400'000u; ++n) {
+        const auto next_square = next_sqrt * next_sqrt;
+
+        const auto is_square = (n == next_square);
+        if (is_square) {
+            ++next_sqrt;
+        }
+
+        EXPECT_EQ(is_perfect_square(n), is_square) << "n = " << n;
+    }
+}
+
+std::vector<uint64_t> strong_lucas_pseudoprimes() {
+    // https://oeis.org/A217255
+    return {5459u,   5777u,   10877u,  16109u,  18971u,  22499u,  24569u,  25199u,  40309u,
+            58519u,  75077u,  97439u,  100127u, 113573u, 115639u, 130139u, 155819u, 158399u,
+            161027u, 162133u, 176399u, 176471u, 189419u, 192509u, 197801u, 224369u, 230691u,
+            231703u, 243629u, 253259u, 268349u, 288919u, 313499u, 324899u};
+}
+
+TEST(LucasDParameter, CanConvertToInt) {
+    EXPECT_EQ(as_int(LucasDParameter{5u, true}), 5);
+    EXPECT_EQ(as_int(LucasDParameter{7u, false}), -7);
+}
+
+TEST(StrongLucas, AllPrimeNumbersAreProbablyPrime) {
+    const auto primes = first_n_primes<3'000u>();
+    for (const auto &p : primes) {
+        if (p > 2u) {
+            EXPECT_THAT(strong_lucas(p), Eq(PrimeResult::PROBABLY_PRIME)) << p;
+        }
+    }
+}
+
+TEST(StrongLucas, GetsFooledByKnownPseudoprimes) {
+    for (const auto &p : strong_lucas_pseudoprimes()) {
+        ASSERT_EQ(miller_rabin(2u, p), PrimeResult::COMPOSITE) << p;
+        EXPECT_THAT(strong_lucas(p), Eq(PrimeResult::PROBABLY_PRIME)) << p;
```

Comment:
A nice side effect of using the matcher syntax is that things tend to align visually.

```suggestion
        ASSERT_THAT(miller_rabin(2u, p), Eq(PrimeResult::COMPOSITE)) << p;
        EXPECT_THAT(strong_lucas(p), Eq(PrimeResult::PROBABLY_PRIME)) << p;
```